greenpeace/planet4-plugin-gutenberg-blocks
# PLANET-6496: HappyPoint override and default content

Ref: https://jira.greenpeace.org/browse/PLANET-6496

Related to: https://github.com/greenpeace/planet4-master-theme/pull/1603

- Use default configuration from Planet 4 > Defaults content
- Add an override option to override the default form type and content

![Screenshot from 2022-02-09 07-53-34](https://user-images.githubusercontent.com/617346/153137534-bfaa5f8a-ddc8-4026-9065-dee33be2c060.png)

## Test

Locally with https://github.com/greenpeace/planet4-master-theme/pull/1603, or [on Pluto](https://www-dev.greenpeace.org/test-pluto/).

- Configure the Defaults content for the Happy Point
- Add a Happy Point block to a page
- The block should use defaults form and url/code unless override is selected
  - On override, the block will use defaults url/code if inputs are left empty
  - With Form type "None", only the background image is used
- Existing blocks from previous version with no "Use mailing list iframe" option checked are converted to "Override + Form type None"
